celer/cython_utils.pyx

```diff
@@ -178,6 +181,8 @@ cdef floating dual_lasso(int n_samples, floating norm_y2,
         d_obj -= (y[i] - n_samples * theta[i]) ** 2
     d_obj *= 0.5 / n_samples
     d_obj += norm_y2 / (2. * n_samples)
+    if l1_ratio != 1.0:
+        d_obj -= alpha * (1 - l1_ratio) * norm_w2
```

Comment:
@mathurinm, since we compute ``dual`` several times within a single loop, I suggest that we stick with this implementation.
To avoid computing ``norm_w2`` in lasso, I will rather condition on ``l1_ratio `` at the beginning of the loop.

Reply:
sounds good